### PR TITLE
refactor(lib/webidl2): modularize attributes/operations

### PR DESCRIPTION
**Pull request by saschanaz**
Not much is remaining now 💪

### CODE REVIEW

File: lib/productions/attribute.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    if (!tokens.base) {
+      tokeniser.unconsume(start_position);
+      return;
+    }
+    ret.idlType = type_with_extended_attributes(tokeniser, "attribute-type") || tokeniser.error("No type in attribute");
+    switch (ret.idlType.generic) {
+      case "sequence":
+      case "record": tokeniser.error(`Attributes cannot accept ${ret.idlType.generic} types`);
+    }
+    tokens.name = tokeniser.consume("identifier", "required") || tokeniser.error("No name in attribute");
```
Comment (marcoscaceres) [inline lib/productions/attribute.js]: I’m a bit unsure what “no name in attribute” means here. Maybe, “attribute lacks a name”?

File: lib/productions/attribute.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    tokens.termination = tokeniser.consume(";") || tokeniser.error("Unterminated attribute");
```
Comment (marcoscaceres) [inline lib/productions/attribute.js]: Maybe add, “, expected ;”

File: lib/productions/operation.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+    if (!special && !regular) {
+      tokens.special = tokeniser.consume("getter", "setter", "deleter");
+    }
+    ret.idlType = return_type(tokeniser) || tokeniser.error("Missing return type");
+    tokens.name = tokeniser.consume("identifier");
+    tokens.open = tokeniser.consume("(") || tokeniser.error("Invalid operation");
+    ret.arguments = argument_list(tokeniser);
+    tokens.close = tokeniser.consume(")") || tokeniser.error("Unterminated operation");
+    tokens.termination = tokeniser.consume(";") || tokeniser.error("Unterminated attribute");
```
Comment (marcoscaceres) [inline lib/productions/operation.js]: Unterminated operation as the error, no? Could be the same as above regarding the “;”.

Comment (saschanaz) [reply lib/productions/operation.js]: Oh wow, never noticed that mistake. Good catch.

**Pull request merged**